### DISCUSSION TITLE
question about parameter(target_cycles)

### DISCUSSION THREAD

**Discussion by MrFaith2001** [fastmachinelearning/hls4ml#1043] [Q&A]
Hello everyone, I checked the related functions for the parameter target cycles, and currently have the following questions:
1. Is target_cycles applicable to the layer of the resource strategy (currently I am testing the conv2d layer under the resource strategy, and there is a big difference between the cycle configured by target_cycles and the actual cycle)?
2. Why is shuffle_cycles 6? How is this number obtained?
function link: [link](https://github.com/fastmachinelearning/hls4ml/blob/ba08ca1737c0b078783d7c2895b40342e5457a47/hls4ml/backends/fpga/fpga_backend.py#L246`)

**Comment by vloncar** [comment]
Hi @MrFaith2001 `target_cycles` was rarely used and may have broken in the meantime. It refers to the `LineBuffer` implementation of convolutional layers when using `io_stream`. 6 cycles were needed to get ready to process one output pixel (in the common case). This was obtained empirically. You can still see a similar number when you synthesize a layer and count up cycles spent outside of the multiplication step.
